### PR TITLE
fix: trim recipient address field to remove leading/ trailing spaces, closes #2806

### LINKED ISSUE #2806: Trim whitespace from address values in send form
a couple of minor things I just noticed with v3.22.1

Sending with an address with a leading or trailing space is not accepted "invalid address" error.
This could confuse some users, so why not strip (commonly pasted) leading and trailing white-space before validating an address?

When I sent a Tx I got a "Nonce conflict" error. But the tx went through and according to the Explorer the nonce was 4 and the prior tx was 3 so it looked fine n not sure why the error msg.

### PR DESCRIPTION
**Pull request by pete-watters**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/wallet/actions/runs/5387403453).

This PR trims the value of the recipient field to remove leading and trailing spaces so that we don't show an error if a user copies in an address with leading/trailing whitespaces

### CODE REVIEW

File: src/app/pages/send/send-crypto-asset-form/components/recipient-field.tsx [hunk 1]
```diff
@@ -22,6 +27,16 @@ export function RecipientField({
   placeholder,
   topInputOverlay,
 }: RecipientFieldProps) {
```
Comment (pete-watters) [inline src/app/pages/send/send-crypto-asset-form/components/recipient-field.tsx:29]: I tried a few approaches and decided on this. I'm not sure if it's the best but it's working well. 

This transforms the value directly and updates the `Formik` field.

I'm open to suggestions of a better way here.

File: tests/specs/send/send-btc.spec.ts [hunk 2]
```diff
@@ -28,6 +28,17 @@ test.describe('send btc', () => {
       const details = await sendPage.confirmationDetails.allInnerTexts();
       test.expect(details).toBeTruthy();
     });
```
Comment (pete-watters) [inline tests/specs/send/send-btc.spec.ts:30]: I added this new test that makes sure the leading and trailing whitespace is removed from the address.

It's almost the same as the following test though ( `test('that asset value and recipient on preview match input'`) so I could merge them. 

That could help us reduce un-necessary test runs. So we would keep the next test but pass in an address with whitespaces

**Review by kyranjamie — APPROVED** [review]
Thanks for fixing and writing a test @pete-watters.

Happy with this approach, but not sure why we need to create the new `useState`? Doesn't it work to just set the field as the trimmed value on it's own?

I see you're comparing, but couldn't you equally compare `field.value === field.value.trim()`?

**Comment by pete-watters** [comment]
> Thanks for fixing and writing a test @pete-watters.
> 
> Happy with this approach, but not sure why we need to create the new `useState`? Doesn't it work to just set the field as the trimmed value on it's own?
> 
> I see you're comparing, but couldn't you equally compare `field.value === field.value.trim()`?

Thanks @kyranjamie! Now that you raised that I don't know why I was doing it either as it's actually not necessary at all. 

I've change the code now so the `trim` will just run when `field.value` updates

**Pull request merged**